### PR TITLE
Decoupling Authentication Flow: Mark IV

### PR DESCRIPTION
**Pull request by jleandroperez**
### Description:
In this PR we're implementing the **WordPressAuthenticatorDelegate** protocol, which, at this stage is responsible of:

- Indicating if a `Support` button should be added at the top right corner of the Login UI.
- Providing a ViewController, to be presented whenever such button is pressed.

@nheagy (Whenever you've got a few spare cycles, next week!!!) would you mind reviewing this one?
Thanks in advance!!!

### Details:
**WordPressAuth.framework** will contain the `WordPressSupportSourceTag` enum, which is a subset of WPiOS's `SupportSourceTag` enum.

Reason: one will live at a framework level, another one will be client app level.

### Testing:
1. Launch WPiOS (fresh install)
2. Verify that the top right button (Help) works as expected, in both, Login and Signup flows.

### CODE REVIEW

**Comment by jleandroperez** [comment]
@ScoutHarris + @nheagy for your consideration!

Thanks in advance!! ❤️

File: WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift [hunk 1]
```diff
@@ -123,11 +129,21 @@ class NUXAbstractViewController: UIViewController, NUXSegueHandler, LoginWithLog
         navigationItem.leftBarButtonItem = cancelButton
     }
 
+    /// Attaches the Help Button, when the WordPressAuthenticationDelegate says we should!
+    ///
+    @objc func setupHelpButtonAndBadgeIfNeeded() {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift]: Does this need `@objc` for sure?

Comment (jleandroperez) [reply WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift]: Nice catch!

**Comment by jleandroperez** [comment]
Thanks for the review @bummytime !!!. Merging this one since BuddyBuild is now 🍏

**Pull request merged**